pkg/capabilities/consensus/ocr3/datafeeds/feeds_aggregator.go

```diff
@@ -140,6 +145,68 @@ func (a *dataFeedsAggregator) Aggregate(previousOutcome *types.AggregationOutcom
 	}, nil
 }
 
+func (a *dataFeedsAggregator) extractSignersAndPayloads(observations map[ocrcommon.OracleID][]values.Value, fConsensus int) ([][]byte, int, map[ocrcommon.OracleID]values.Value) {
```

Comment:
there is an awful lot of control logic in this func. can you break it into a helper or two. specifically, the inner for loops the `breaks` are hard to read, because i have to think about which loop is being broken.

Reply:
Moved the inner loop to a helper.